### DISCUSSION TITLE
umi4 Invalid config keys: locale

### DISCUSSION THREAD

**Discussion by weilun0510** [umijs/umi#9100] [Q&A] [answered]
umi4 I want to add multilingual configuration. When I add locale in .umirc.ts I get "Invalid config keys: locale"

**Comment by fz6m** [comment, marked as answer]
1. 创建 max 项目，内置了 locale 功能

2. 或者单独安装启用 i18n 插件：

    ```bash
      pnpm add @umijs/plugins
    ```
    
    ```ts
    // .umirc.ts
    
    export default {
      plugins: ['@umijs/plugins/dist/locale'],
      locale: {}
    }
    ```

FYI :

 - https://umijs.org/docs/max/i18n

**Reply by weilun0510** [reply]
Thanks a lot, this worked for me. I can't find the instructions for installing and enabling the i18n plugin in the official documentation, where did you find it?

**Reply by weilun0510** [reply]
请问 `plugins: ['@umijs/plugins/dist/locale'],` 这个配置是哪里看到的呢？我现在想开启request，但是找不到文档

**Reply by fz6m** [reply]
所有 max 中的插件都可以在 `@umijs/plugins/dist/${plugin-name}` 中导入。这些插件在 max 中内置，如果需要多个，更推荐直接使用 max 项目，无需手动写 `@umijs/plugins/dist/${plugin-name}`

**Reply by weilun0510** [reply]
谢谢。我创建模板的时候没有选择max，还能升级为max吗

**Reply by fz6m** [reply]
需要人工处理的比较多：

1. `package.json` 中 `umi` 依赖替换为 `@umijs/max` ，`scripts` 命令中 `umi` 替换为 `max` 。

2. 所有代码从 `umi` 导入的语句都要换成从 `@umijs/max` 中导入。

如需迁移，可以初始化一个新 max 项目对照改一下。

如果觉得麻烦，就自己写 `@umijs/plugins/dist/${plugin-name}` 吧：

```ts
// .umirc.ts

export default {
  plugins: [
    '@umijs/plugins/dist/request',
    '@umijs/plugins/dist/locale',
    '@umijs/plugins/dist/model',
    // ... you need plugins
  ],
  locale: {},
  request: {},
  // ...
}
```

`max` 和 `umi` 的区别只有插件内置的区别，想用哪个手动声明就可以了。

**Reply by weilun0510** [reply]
好的，谢谢
